### PR TITLE
Cors obs

### PR DESCRIPTION
**Pull request by denniszollo**
need new obs message.  After merge need a new veresion and pypi release.

Note that scaling has changed on pseudorange from 1 cm to 2cm for count.

Will require update to anything using sbp eventually:
piksi_tools (done in PR: https://github.com/swift-nav/piksi_tools/pull/273)
sbp_log_analysis (need to do)
gnss_analysis (need to do)

/cc @mookerji @mfine @JoshuaGross  any time to review?

### CODE REVIEW

**Comment by mookerji** [comment]
Yeah, looking.

**Comment by mookerji** [comment]
I'd say that we update the descriptions as well to describe why these messages have changed. Then we'll have to 1. release the library, 2. update the tools, 3. package release candidate for tools, 4. release firmware.

File: haskell/src/SwiftNav/SBP/Observation.hs [hunk 1]
```diff
@@ -686,6 +690,44 @@ $(deriveJSON defaultOptions {fieldLabelModifier = fromMaybe "_packedObsContentDe
...
 $(makeLenses ''PackedObsContentDepA)
 
+-- | PackedObsContentDepB.
+--
+-- Pseudorange and carrier phase observation for a satellite being tracked.
+-- Pseudoranges are referenced to a nominal pseudorange.
+data PackedObsContentDepB = PackedObsContentDepB
+  { _packedObsContentDepB_P  :: Word32
+    -- ^ Pseudorange observation
+  , _packedObsContentDepB_L  :: CarrierPhase
```
Comment (ljbade) [inline haskell/src/SwiftNav/SBP/Observation.hs]: Can we add note on carrier phase sign? Also should have units

File: haskell/src/SwiftNav/SBP/Observation.hs [hunk 2]
```diff
@@ -686,6 +690,44 @@ $(deriveJSON defaultOptions {fieldLabelModifier = fromMaybe "_packedObsContentDe
              ''PackedObsContentDepA)
 $(makeLenses ''PackedObsContentDepA)
 
+-- | PackedObsContentDepB.
+--
+-- Pseudorange and carrier phase observation for a satellite being tracked.
+-- Pseudoranges are referenced to a nominal pseudorange.
+data PackedObsContentDepB = PackedObsContentDepB
+  { _packedObsContentDepB_P  :: Word32
```
Comment (ljbade) [inline haskell/src/SwiftNav/SBP/Observation.hs]: Units

File: haskell/src/SwiftNav/SBP/Observation.hs [hunk 3]
```diff
@@ -686,6 +690,44 @@ $(deriveJSON defaultOptions {fieldLabelModifier = fromMaybe "_packedObsContentDe
...
+    -- ^ Pseudorange observation
+  , _packedObsContentDepB_L  :: CarrierPhase
+    -- ^ Carrier phase observation
+  , _packedObsContentDepB_cn0 :: Word8
```
Comment (ljbade) [inline haskell/src/SwiftNav/SBP/Observation.hs]: Units

File: haskell/src/SwiftNav/SBP/Observation.hs (hunk 1 above)
Comment (denniszollo) [inline haskell/src/SwiftNav/SBP/Observation.hs]: This is a type, I can't comment about the units.

File: haskell/src/SwiftNav/SBP/Observation.hs (hunk 2 above)
Comment (denniszollo) [inline haskell/src/SwiftNav/SBP/Observation.hs]: same- the units are some weird fixed point format.

File: spec/yaml/swiftnav/sbp/observation.yaml [hunk 4]
```diff
@@ -33,22 +33,6 @@ definitions:
...
-        - i:
-            type: s32
-            units: cycles
-            desc: Carrier phase whole cycles
-        - f:
-            type: u8
-            units: cycles / 256
-            desc: Carrier phase fractional part
-
  - ObservationHeader:
```
Comment (mfine) [inline spec/yaml/swiftnav/sbp/observation.yaml]: You're changing the contents of `ObservationHeader` but continuing to use it from the deprecated messages. You'll need a deprecated `ObservationHeader` structure s well to preserve the functioning of the deprecated messages.

File: spec/yaml/swiftnav/sbp/observation.yaml [hunk 5]
```diff
@@ -543,4 +588,150 @@ definitions:
...
+        - obs:
+            type: array
+            fill: PackedObsContentDepB
+            map_by: sid
+            desc: |
+              Pseudorange and carrier phase observation for a
+              satellite being tracked.
 
+ - MSG_EPHEMERIS_DEP_C:
+    id: 0x0047
```
Comment (mfine) [inline spec/yaml/swiftnav/sbp/observation.yaml]: Bummer if this is only for an `id` change to get it in a different mask.

File: spec/yaml/swiftnav/sbp/observation.yaml (hunk 4 above)
Comment (denniszollo) [inline spec/yaml/swiftnav/sbp/observation.yaml]: I didn't change the header.....

Comment (mfine) [inline spec/yaml/swiftnav/sbp/observation.yaml]: Woops, missed the separator between the next set of changes for `PackedObsContent`. Sorry.

Comment (mookerji) [inline spec/yaml/swiftnav/sbp/observation.yaml]: @mfine, neither the interface nor content of the fields is changing. What is changing is the way those fields in the observations are computed, so I don't think that needs to change.

Comment (mfine) [inline spec/yaml/swiftnav/sbp/observation.yaml]: Yeah, I don't think it needs to change - I misread the diff and incorporated `PackedObsContent` changes into the observation header structure.

File: spec/yaml/swiftnav/sbp/observation.yaml (hunk 5 above)
Comment (denniszollo) [inline spec/yaml/swiftnav/sbp/observation.yaml]: It is only for hte ID, but I have had big problems not being able to mask of the comparatively large ephemeris message so there is value in it.

**Comment by denniszollo** [comment]
This is ready but is still waiting on a few more unit tests after we have compatible firmware.

**Comment by denniszollo** [comment]
deprecated in favor of : https://github.com/swift-nav/libsbp/pull/356

**Pull request closed without merging**